auth/src/main/java/com/firebase/ui/auth/ui/phone/PhoneActivity.java

```diff
@@ -83,7 +84,18 @@ protected void onFailure(@NonNull Exception e) {
             protected void onSuccess(@NonNull PhoneVerification verification) {
                 if (verification.isAutoVerified()) {
                     Toast.makeText(
-                            PhoneActivity.this, R.string.fui_auto_verified, Toast.LENGTH_LONG).show();
+                            PhoneActivity.this,
+                            R.string.fui_auto_verified,
+                            Toast.LENGTH_LONG
+                    ).show();
+
+                    FragmentManager manager = getSupportFragmentManager();
+                    if (manager.findFragmentByTag(SubmitConfirmationCodeFragment.TAG) != null) {
+                        // Ensure the submit code screen isn't visible if there's no code to submit.
+                        // It's possible to get into this state when an SMS is sent, but then
+                        // automatically retrieved.
+                        manager.popBackStack();
```

Comment:
This works for auto-verification (where no SMS is sent) but does not work for the case where the code is sent but then auto-retrieved.  In that case we get `onCodeSent` first and then when this happens the flow goes a little crazy, I couldn't capture a video of it.

Reply:
I think I fixed the craziness, but I'm still not super happy with the UX. It's either show the Smart Lock dialog on top of a code-less edittext or show it over the phone number entry fragment. I know we have a toast, but maybe we need a brief bottom sheet or something? Either way, I doubt I'll have the time to play with that. 😢